canvas2svg.js

```diff
@@ -461,33 +468,38 @@
     /**
      * Sets current element to parent, or just root if already root
      */
-    ctx.prototype.restore = function(){
+    ctx.prototype.restore = function () {
         this.__currentElement = this.__groupStack.pop();
+        this.__currentElementsToStyle = null;
         //Clearing canvas will make the poped group invalid, currentElement is set to the root group node.
         if (!this.__currentElement) {
             this.__currentElement = this.__root.childNodes[1];
         }
         var state = this.__stack.pop();
         this.__applyStyleState(state);
-
     };
 
     /**
      * Helper method to add transform
      * @private
      */
-    ctx.prototype.__addTransform = function(t) {
-
+    ctx.prototype.__addTransform = function (t) {
         //if the current element has siblings, add another group
         var parent = this.__closestGroupOrSvg();
-        if(parent.childNodes.length > 0) {
+        if (parent.childNodes.length > 0) {
+        	if (this.__currentElement.nodeName === "path") {
+        		this.__currentElementsToStyle || (this.__currentElementsToStyle = { element: parent, children: [] });
```

Comment:
@mudcube This is the line that @webspinner requested changing to a regular if statement. Please let me know if you have to time to make the changes, otherwise I may supercede your PR with another one.